components/RedditProvider.tsx

```diff
@@ -7,6 +7,8 @@ export interface RedditProviderProps {
   subReddit: any
   setSort: (sort: string) => void
   setSubreddit: (subReddit: {}) => void
+  searchInput: string
+  setSearchInput: React.Dispatch<React.SetStateAction<string>>
```

Comment:
@AloyBranCheung While I appreciate the verboseness of this type, I'd rather it stay in line with the rest of the project. Can you update it to something like: `(searchInput: string) => void` 

Thank you